server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -279,14 +276,14 @@ private void finalizeReplication(CheckpointInfoResponse checkpointInfoResponse)
             // In this case the shard is closed at some point while updating the reader.
             // This can happen when the engine is closed in a separate thread.
             logger.warn("Shard is already closed, closing replication");
-        } catch (OpenSearchException ex) {
+        } catch (CancellableThreads.ExecutionCancelledException ex) {
```

Comment:
note - I've cleaned up the handling here.  This path was incorrectly assuming any OSE was a cancellation and anything that was not an OSE was a corruption error.

It will now 1. Check for corruption and handle it by throwing a corruption exception.  2. Check for a cancellation and handle it (swallow the error so its not reported as a failure) or 3. throw it as a regular failure exception.  Anything not explicitly caught as corruption will be bubbled as a replication failed (non catastrophic/shard failure) exception.